### PR TITLE
feat(alert): support actions-end

### PR DESCRIPTION
**Pull request by Elijbet**
**Related Issue:** #2382

## Summary

`Alert` now supports `actions-end` slot for adding actions to the end of the component, with a recommendation to use two or fewer actions.

### CODE REVIEW

File: src/components/alert/alert.tsx [hunk 1]
```diff
@@ -201,6 +203,9 @@ export class Alert implements OpenCloseComponent, LocalizedComponent {
     const { active, autoDismiss, label, placement, queued, requestedIcon } = this;
     const role = autoDismiss ? "alert" : "alertdialog";
     const hidden = !active;
+
+    const hasActionEnd = getSlotted(el, SLOTS.actionsEnd);
```
[the remark this answers is not in the document]
Comment (driskull) [reply src/components/alert/alert.tsx:207]: Example usage: https://github.com/Esri/calcite-components/blob/master/src/components/panel/panel.tsx#L482

File: src/components/alert/alert.tsx [hunk 2]
```diff
@@ -226,7 +231,12 @@ export class Alert implements OpenCloseComponent, LocalizedComponent {
             <slot name={SLOTS.message} />
             <slot name={SLOTS.link} />
           </div>
-          {queueCount}
+          {hasActionEnd ? (
+            <div class={CSS.actionsEnd}>
```
Comment (jcfranco) [inline src/components/alert/alert.tsx:235]: Per [Stencil recommendation](https://stenciljs.com/docs/templating-jsx#dealing-with-children), can you add `key` to any conditionally rendered nodes?

File: src/components/alert/alert.tsx [hunk 3]
```diff
@@ -237,7 +248,10 @@ export class Alert implements OpenCloseComponent, LocalizedComponent, LoadableCo
             <slot name={SLOTS.message} />
             <slot name={SLOTS.link} />
           </div>
-          {queueCount}
+          <div class={CSS.actionsEnd} hidden={!hasEndActions}>
```
Comment (driskull) [inline src/components/alert/alert.tsx:251]: I think you will need to add the following CSS...

```css
[hidden] {
  @apply hidden;
}
```

I needed this for the `calcite-panel` to work correctly.

https://github.com/Esri/calcite-components/pull/5129#discussion_r942875228

File: src/components/alert/alert.tsx [hunk 4]
```diff
@@ -413,4 +429,12 @@ export class Alert implements OpenCloseComponent, LocalizedComponent, LoadableCo
     window.clearTimeout(this.queueTimeout);
     this.queueTimeout = window.setTimeout(() => (this.queued = false), 300);
   }
+
+  handleActionsEndSlotChange = (event: Event): void => {
```
Comment (driskull) [inline src/components/alert/alert.tsx:433]: You can remove this function in favor of this dom utility.

https://github.com/Esri/calcite-components/blob/3eb037e627832c34334969d0dc0cf5dde3ce42bc/src/utils/dom.ts#L315-L317

**Review by driskull — APPROVED** [review]
Looks good after a couple changes :) 👍

**Pull request merged**